### ISSUE TITLE
Infinite loop in BrotliEncoder

### ISSUE THREAD

**Issue by kimk-s** [Cysharp/MemoryPack#279]
Hello!

It seems like an infinite loop is occurring in BrotliCompressor while my program is running.
I can't reproduce this exactly, but I did my best.
If you don't mind, I would appreciate it if you could check the code below.

Used MemoryPack version: 1.21.0

```csharp
var buffer = new ArrayBufferWriter<byte>(256);
using var state = MemoryPackWriterOptionalStatePool.Rent(null);
var writer = new MemoryPackWriter<ArrayBufferWriter<byte>>(ref buffer, state);

var compressor = new BrotliCompressor();
var coWriter = new MemoryPackWriter<BrotliCompressor>(ref compressor, writer.OptionalState);

var bytes = new byte[249];
Random.Shared.NextBytes(bytes);
coWriter.WriteUnmanagedArray(bytes);
coWriter.Flush();

compressor.CopyTo(ref writer);
```

**Comment by neuecc** [comment]
thanks for the report, I'll check soon.

**Comment by neuecc** [comment]
Thanks, I've fixed at v1.21.1.

**Comment by NicoIer** [comment]
> Thanks, I've fixed at v1.21.1.
1.21.1 still has this problem, it's still stuck
<img width="810" alt="image" src="https://github.com/user-attachments/assets/bcae6b04-5fb0-483b-b7f2-bcf22a09d0ca">
